tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/AbstractCamelRouteITest.java

```diff
@@ -116,6 +117,9 @@ public Option[] config() {
         if (hasCamelRouteSupplierFilter()) {
             combine = combine(combine, getCamelRouteSupplierFilter());
         }
+        if (hasCamekKarafTestHint()) {
+            combine = combine(combine, getCamelRouteIgnoreSupplier());
+        }
```

Comment:
```suggestion
        if (hasCamekKarafTestHint()) {
            combine = combine(combine, getCamelRouteIgnoreSupplier());
        } else if (hasCamelRouteSupplierFilter()) {
            combine = combine(combine, getCamelRouteSupplierFilter());
        }
```
They are exclusive, it can only be one or the other

Reply:
Done